Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Beckn/Select.hs

```diff
@@ -164,4 +190,25 @@ buildSearchReqLocation merchantId sessionToken _ latLong@Maps.LatLong {..} = do
   now <- getCurrentTime
   let createdAt = now
       updatedAt = now
-  pure DLoc.SearchReqLocation {..}
+  pure DLoc.SearchReqLocation {lat = latLong.lat, lon = latLong.lon, ..}
+
+decodeAddress :: BA.Address -> Maybe Text
+decodeAddress address = do
+  if removeSpace (fromJust address.city) == "" && removeSpace (fromJust address.locality) == "" && removeSpace (fromJust address.street) == "" && removeSpace (fromJust address.door) == "" && removeSpace (fromJust address.building) == ""
```

Comment:
you shouldn't use `fromJust` in production code
and also I think this code can ber refactored somehow, because it has many repeats

Reply:
can you suggest how can this be done in a better way in haskell?

Reply:
@umangj-28 what is expected behaivoir of this function?
door, building, street, locality, city, state, area_code, country

Reply:
Yeah this is the order: `door, building, street, locality, city, state area_code, country`
we are showing `state area_code, country` always, and all other Just fields get appended with a comma

Reply:
I can see some little difference from you variant, but I think this also can be handled in my way

Reply:
Sure I'll test it, thanks @roman-bodavskiy